### DISCUSSION TITLE
Quick focus tool in zospy?

### DISCUSSION THREAD

**Discussion by BildIngenieur** [MREYE-LUMC/ZOSPy#76] [Q&A]
Hello,

Is there a way to use the quick focus tool in zospy?
I came across this for optimization: https://github.com/MREYE-LUMC/ZOSPy/discussions/72 but when I try to use the same approach for quick focus it fails at the step `quickFocus = oss.Tools.OpenQuickFocus()` with the error `AttributeError: 'bool' object has no attribute 'Tools'`.

**Comment by LucVV** [comment]
Hi @BildIngenieur,

I can't check if `oss.Tools.OpenQuickFocus()`  works, but it should as far as I know.

Based on your question, I assume the error happens before. In your code, `oss` seems to be a Boolean rather than an `OpticStudioSystem` instance. 

My guess is that you are using one of the the legacy methods to initiate the connection (`zos.connect_as_extension()`, `zos.create_new_application()` or `zos.connect_as_standalone()`). These methods have been deprecated since version 1.2.0. If you are indeed using one of these, the preffered option is to change to [`zos.connect()`](https://zospy.readthedocs.io/en/latest/api/zospy.zpcore.ZOS.html#zospy.zpcore.ZOS.connect) as the legacy methods will be removed in the future. 

Does that solve your issue? If not, can you paste a code snippet of how you initiate the connection as well as provide your version of ZOSPy?
